File: onepg/lib/onepg/sandwich_buttons.ex
defmodule Onepg.SandwichButtons do
  @button_nodes ["<button onclick=\"document.body.background = ''\">Sandwich Off</button>", "<button onclick=\"document.body.background = 'sandwich.jpg'\">Sandwich On</button>\n"]

  defstruct [html: Enum.join(@button_nodes, "\n"), nodes: @button_nodes]
end

File: docker-assets/onepg/make_the_page/lib/sandwich_buttons.rb
require_relative 'html_document'

class SandwichButtons
  attr_reader :button_nodes, :buttons_html, :html_document

  def initialize(document = HtmlDocument.new)
    @html_document = document
    @button_nodes = [
       "<button onclick=\"document.body.background = ''\">Sandwich Off</button>\n",
       "<button onclick=\"document.body.background = 'sandwich.jpg'\">Sandwich On</button>\n"
      ]

    @html_document.add_nodes(button_nodes)
    @buttons_html = @button_nodes.join
  end
end

